src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -1569,6 +1570,34 @@ public function getContent(bool $asResource = false)
         return $this->content;
     }
 
+    /**
+     * Gets the request body decoded as array, typically from a JSON payload.
+     *
+     * @throws JsonException When the body cannot be decoded to an array
+     */
+    public function toArray(): array
+    {
+        if ('' === $content = $this->getContent()) {
+            throw new JsonException('Response body is empty.');
+        }
+
+        try {
+            $content = json_decode($content, true, 512, \JSON_BIGINT_AS_STRING | (\PHP_VERSION_ID >= 70300 ? \JSON_THROW_ON_ERROR : 0));
+        } catch (\JsonException $e) {
+            throw new JsonException('Could not decode request body.', $e->getCode(), $e);
+        }
+
+        if (\PHP_VERSION_ID < 70300 && \JSON_ERROR_NONE !== json_last_error()) {
+            throw new JsonException(json_last_error_msg(), json_last_error());
+        }
+
+        if (!\is_array($content)) {
```

Comment:
But what about object of JSON. JSON scheme is not only array of something, but can be as object.

Reply:
Correct, But this method is called `toArray()`. That is why I return an array. =)

Reply:
Well, yes - I return my honor.